pandas/core/generic.py

```diff
@@ -3172,20 +3172,15 @@ def _align_series(self, other, join='outer', axis=None, level=None,
 
         else:
 
-            # for join compat if we have an unnamed index, but
-            # are specifying a level join
-            other_index = other.index
-            if level is not None and other.index.name is None:
```

Comment:
As far as I understand the code, this *was* hit, and this fixed the issue in the PR I linked, but it only fixed it for the case when you provide real level names for `level` (in this case, `other` is given the same name, and then `Index.join` will automatically join on the common level name. 
When passing integer number to `level`, setting this number as the level name does not ensure that `Index.join` can join it correctly, as there are no common level names.

Another option is to set here the correct level name from `self` that corresponds to the provided level number, but as passing `level=level` to join also works (for both cases) and is simpler, I took that option.